src/lib/pickles_types/plonk_types.ml

```diff
@@ -167,8 +167,122 @@ module Opt = struct
   end
 end
 
-module Lookup_config = struct
-  type t = { lookup : Opt.Flag.t; runtime : Opt.Flag.t }
+module Features = struct
+  [%%versioned
+  module Stable = struct
+    module V1 = struct
+      type 'bool t =
+        { chacha : 'bool
+        ; range_check : 'bool
+        ; foreign_field_add : 'bool
+        ; foreign_field_mul : 'bool
+        ; xor : 'bool
+        ; rot : 'bool
+        ; lookup : 'bool
+        ; runtime_tables : 'bool
+        }
+      [@@deriving sexp, compare, yojson, hash, equal, hlist]
+    end
+  end]
+
+  let to_vector
+      { chacha
+      ; range_check
+      ; foreign_field_add
+      ; foreign_field_mul
+      ; xor
+      ; rot
+      ; lookup
+      ; runtime_tables
+      } : _ Vector.t =
+    [ chacha
+    ; range_check
+    ; foreign_field_add
+    ; foreign_field_mul
+    ; xor
+    ; rot
+    ; lookup
+    ; runtime_tables
+    ]
+
+  let of_vector
+      ([ chacha
+       ; range_check
+       ; foreign_field_add
+       ; foreign_field_mul
+       ; xor
+       ; rot
+       ; lookup
+       ; runtime_tables
+       ] :
+        _ Vector.t ) =
+    { chacha
+    ; range_check
+    ; foreign_field_add
+    ; foreign_field_mul
+    ; xor
+    ; rot
+    ; lookup
+    ; runtime_tables
+    }
+
+  let typ bool =
+    Vector.typ bool Nat.N8.n
+    |> Snarky_backendless.Typ.transport ~there:to_vector ~back:of_vector
+    |> Snarky_backendless.Typ.transport_var ~there:to_vector ~back:of_vector
+
+  let none =
+    { chacha = Opt.Flag.No
+    ; range_check = Opt.Flag.No
+    ; foreign_field_add = Opt.Flag.No
+    ; foreign_field_mul = Opt.Flag.No
+    ; xor = Opt.Flag.No
+    ; rot = Opt.Flag.No
+    ; lookup = Opt.Flag.No
+    ; runtime_tables = Opt.Flag.No
+    }
+
+  let none_bool =
```

Comment:
Suggestion: Maybe call this `empty_feature_flags`, `no_feature_flags, `all_false`, `all_features_off`?